from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1300, 800)
        MainWindow.setAutoFillBackground(True)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.scroll_area = QtWidgets.QScrollArea(self.centralwidget)
        self.scroll_area.setEnabled(True)
        self.scroll_area.setGeometry(QtCore.QRect(280, 120, 1015, 620))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.scroll_area.sizePolicy().hasHeightForWidth())
        self.scroll_area.setSizePolicy(sizePolicy)
        self.scroll_area.setMinimumSize(QtCore.QSize(0, 0))
        self.scroll_area.setSizeIncrement(QtCore.QSize(0, 0))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.scroll_area.setFont(font)
        self.scroll_area.setFrameShape(QtWidgets.QFrame.Box)
        self.scroll_area.setFrameShadow(QtWidgets.QFrame.Plain)
        self.scroll_area.setLineWidth(3)
        self.scroll_area.setMidLineWidth(0)
        self.scroll_area.setVerticalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
        self.scroll_area.setHorizontalScrollBarPolicy(QtCore.Qt.ScrollBarAlwaysOff)
        self.scroll_area.setSizeAdjustPolicy(QtWidgets.QAbstractScrollArea.AdjustToContents)
        self.scroll_area.setWidgetResizable(False)
        self.scroll_area.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignTop)
        self.scroll_area.setObjectName("scroll_area")
        self.scrollAreaWidgetContents = QtWidgets.QWidget()
        self.scrollAreaWidgetContents.setGeometry(QtCore.QRect(0, 0, 966, 609))
        self.scrollAreaWidgetContents.setObjectName("scrollAreaWidgetContents")
        self.add_test = QtWidgets.QPushButton(self.scrollAreaWidgetContents)
        self.add_test.setGeometry(QtCore.QRect(10, 10, 970, 50))
        self.add_test.setMinimumSize(QtCore.QSize(970, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.add_test.setFont(font)
        self.add_test.setObjectName("add_test")
        self.add_stu = QtWidgets.QPushButton(self.scrollAreaWidgetContents)
        self.add_stu.setGeometry(QtCore.QRect(10, 10, 970, 50))
        self.add_stu.setMinimumSize(QtCore.QSize(970, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.add_stu.setFont(font)
        self.add_stu.setObjectName("add_stu")
        self.sort_btn = QtWidgets.QPushButton(self.scrollAreaWidgetContents)
        self.sort_btn.setGeometry(QtCore.QRect(10, 10, 970, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.sort_btn.setFont(font)
        self.sort_btn.setObjectName("sort_btn")
        self.score = QtWidgets.QLabel(self.scrollAreaWidgetContents)
        self.score.setGeometry(QtCore.QRect(10, 70, 970, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.score.setFont(font)
        self.score.setLineWidth(1)
        self.score.setObjectName("score")
        self.scroll_area.setWidget(self.scrollAreaWidgetContents)
        self.choose_class = QtWidgets.QComboBox(self.centralwidget)
        self.choose_class.setGeometry(QtCore.QRect(280, 30, 810, 40))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.choose_class.setFont(font)
        self.choose_class.setObjectName("choose_class")
        self.main = QtWidgets.QPushButton(self.centralwidget)
        self.main.setGeometry(QtCore.QRect(30, 30, 170, 40))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.main.setFont(font)
        self.main.setObjectName("main")
        self.add_class = QtWidgets.QPushButton(self.centralwidget)
        self.add_class.setGeometry(QtCore.QRect(1230, 30, 40, 40))
        self.add_class.setText("")
        self.add_class.setObjectName("add_class")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(30, 80, 231, 31))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.plainTextEdit = QtWidgets.QPlainTextEdit(self.centralwidget)
        self.plainTextEdit.setGeometry(QtCore.QRect(30, 120, 230, 620))
        self.plainTextEdit.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.plainTextEdit.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.plainTextEdit.setLineWidth(1)
        self.plainTextEdit.setObjectName("plainTextEdit")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(280, 80, 991, 31))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.label_2.setFont(font)
        self.label_2.setLineWidth(1)
        self.label_2.setObjectName("label_2")
        self.info = QtWidgets.QPushButton(self.centralwidget)
        self.info.setGeometry(QtCore.QRect(1110, 30, 40, 40))
        self.info.setText("")
        self.info.setObjectName("info")
        self.del_class = QtWidgets.QPushButton(self.centralwidget)
        self.del_class.setGeometry(QtCore.QRect(1170, 30, 40, 40))
        self.del_class.setText("")
        self.del_class.setObjectName("del_class")
        font = QtGui.QFont()
        font.setPointSize(15)
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1300, 18))
        self.menubar.setObjectName("menubar")
        self.menu = QtWidgets.QMenu(self.menubar)
        self.menu.setObjectName("menu")
        self.menu_2 = QtWidgets.QMenu(self.menubar)
        self.menu_2.setObjectName("menu_2")
        self.menu_3 = QtWidgets.QMenu(self.menubar)
        self.menu_3.setObjectName("menu_3")
        self.set_lang = QtWidgets.QPushButton(self.centralwidget)
        self.set_lang.setObjectName("set_lang")
        self.set_lang.setGeometry(QtCore.QRect(220, 30, 40, 40))
        self.set_lang.setText("")
        self.set_lang.setObjectName("set_lang")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.set_font = QtWidgets.QAction(MainWindow)
        self.set_font.setObjectName("set_font")
        self.write_Python = QtWidgets.QAction(MainWindow)
        self.write_Python.setObjectName("write_Python")
        self.rand_stu = QtWidgets.QAction(MainWindow)
        self.rand_stu.setEnabled(True)
        self.rand_stu.setObjectName("rand_stu")
        self.write_html = QtWidgets.QAction(MainWindow)
        self.write_html.setObjectName("write_html")
        self.set_bg = QtWidgets.QAction(MainWindow)
        self.set_bg.setObjectName("set_bg")
        self.set_btn = QtWidgets.QAction(MainWindow)
        self.set_btn.setObjectName("set_btn")
        self.menu.addAction(self.rand_stu)
        self.menu.addAction(self.write_html)
        self.menu_2.addAction(self.write_Python)
        self.menu_3.addAction(self.set_font)
        self.menu_3.addAction(self.set_bg)
        self.menu_3.addAction(self.set_btn)
        self.menubar.addAction(self.menu.menuAction())
        self.menubar.addAction(self.menu_2.menuAction())
        self.menubar.addAction(self.menu_3.menuAction())

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "ECH 课堂助手"))
        self.add_test.setText(_translate("MainWindow", "添加测试"))
        self.add_stu.setText(_translate("MainWindow", "添加学生"))
        self.sort_btn.setText(_translate("MainWindow", "按成绩排序"))
        self.score.setText(_translate("MainWindow", "平均分："))
        self.main.setText(_translate("MainWindow", "主菜单"))
        self.add_class.setToolTip(_translate("MainWindow", "添加班级"))
        self.label.setText(_translate("MainWindow", "记事本"))
        self.label_2.setText(_translate("MainWindow", "该班级的 测试/学生"))
        self.info.setToolTip(_translate("MainWindow", "班级信息"))
        self.del_class.setToolTip(_translate("MainWindow", "删除班级"))
        self.menu.setTitle(_translate("MainWindow", "基础 (&B)"))
        self.menu_2.setTitle(_translate("MainWindow", "高级 (&A)"))
        self.menu_3.setTitle(_translate("MainWindow", "设置 (&S)"))
        self.set_font.setText(_translate("MainWindow", "设置字体 (&F)"))
        self.set_font.setShortcut(_translate("MainWindow", "Ctrl+Alt+F"))
        self.write_Python.setText(_translate("MainWindow", "编写Python代码 (&P)"))
        self.write_Python.setShortcut(_translate("MainWindow", "Ctrl+Alt+P"))
        self.rand_stu.setText(_translate("MainWindow", "随机点名 (&R)"))
        self.rand_stu.setShortcut(_translate("MainWindow", "Ctrl+Alt+R"))
        self.write_html.setText(_translate("MainWindow", "编写文档 (&H)"))
        self.write_html.setShortcut(_translate("MainWindow", "Ctrl+Alt+H"))
        self.set_bg.setText(_translate("MainWindow", "设置背景 (&G)"))
        self.set_bg.setShortcut(_translate("MainWindow", "Ctrl+Alt+G"))
        self.set_btn.setText(_translate("MainWindow", "设置按钮颜色 (&C)"))
        self.set_btn.setShortcut(_translate("MainWindow", "Ctrl+Alt+B"))


class Html_Dialog(object):
    def setupUi(self, htmlDialog):
        htmlDialog.setObjectName("htmlDialog")
        htmlDialog.setFixedSize(1000, 750)
        self.textEdit = QtWidgets.QTextEdit(htmlDialog)
        self.textEdit.setGeometry(QtCore.QRect(10, 70, 980, 670))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.textEdit.setFont(font)
        self.textEdit.setObjectName("textEdit")
        self.save = QtWidgets.QPushButton(htmlDialog)
        self.save.setGeometry(QtCore.QRect(10, 10, 50, 50))
        self.save.setText("")
        self.save.setObjectName("save")
        self.save.setToolTip("保存")
        self.open = QtWidgets.QPushButton(htmlDialog)
        self.open.setGeometry(QtCore.QRect(70, 10, 50, 50))
        self.open.setText("")
        self.open.setObjectName("open")
        self.open.setToolTip("打开")
        self.lineEdit = QtWidgets.QLineEdit(htmlDialog)
        self.lineEdit.setGeometry(QtCore.QRect(130, 10, 600, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.lineEdit.setFont(font)
        self.lineEdit.setText("")
        self.lineEdit.setObjectName("lineEdit")
        self.see = QtWidgets.QPushButton(htmlDialog)
        self.see.setGeometry(QtCore.QRect(740, 10, 250, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.see.setFont(font)
        self.see.setObjectName("see")

        self.retranslateUi(htmlDialog)
        QtCore.QMetaObject.connectSlotsByName(htmlDialog)

    def retranslateUi(self, htmlDialog):
        _translate = QtCore.QCoreApplication.translate
        htmlDialog.setWindowTitle(_translate("htmlDialog", "编写HTML文档"))
        self.lineEdit.setPlaceholderText(_translate("htmlDialog", "请输入保存的文件名"))
        self.see.setText(_translate("htmlDialog", "预览文本内容"))
        self.save.setToolTip(_translate("htmlDialog","保存"))
        self.open.setToolTip(_translate("htmlDialog","打开"))


class Code_Dialog(object):
    def setupUi(self, codeDialog):
        codeDialog.setObjectName("codeDialog")
        codeDialog.setFixedSize(1000, 750)
        self.save = QtWidgets.QPushButton(codeDialog)
        self.save.setGeometry(QtCore.QRect(10, 10, 50, 50))
        self.save.setText("")
        self.save.setObjectName("save")
        self.open = QtWidgets.QPushButton(codeDialog)
        self.open.setGeometry(QtCore.QRect(70, 10, 50, 50))
        self.open.setText("")
        self.open.setObjectName("open")
        self.lineEdit = QtWidgets.QLineEdit(codeDialog)
        self.lineEdit.setGeometry(QtCore.QRect(130, 10, 860, 50))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.lineEdit.setFont(font)
        self.lineEdit.setText("")
        self.lineEdit.setDragEnabled(False)
        self.lineEdit.setObjectName("lineEdit")
        self.plainTextEdit = QtWidgets.QPlainTextEdit(codeDialog)
        self.plainTextEdit.setFont(font)
        self.plainTextEdit.setGeometry(QtCore.QRect(10, 70, 980, 670))
        self.plainTextEdit.setObjectName("plainTextEdit")

        self.retranslateUi(codeDialog)
        QtCore.QMetaObject.connectSlotsByName(codeDialog)

    def retranslateUi(self, codeDialog):
        _translate = QtCore.QCoreApplication.translate
        codeDialog.setWindowTitle(_translate("codeDialog", "编写Python代码"))
        self.save.setToolTip(_translate("codeDialog", "保存并运行"))
        self.open.setToolTip(_translate("codeDialog", "打开"))
        self.lineEdit.setPlaceholderText(_translate("codeDialog", "请输入保存的文件名"))


class Color_Dialog(object):
    def setupUi(self, colorDialog):
        colorDialog.setObjectName("colorDialog")
        colorDialog.setFixedSize(400, 300)
        self.buttonBox = QtWidgets.QDialogButtonBox(colorDialog)
        self.buttonBox.setGeometry(QtCore.QRect(20, 250, 350, 30))
        self.buttonBox.setStandardButtons(QtWidgets.QDialogButtonBox.Cancel|QtWidgets.QDialogButtonBox.Ok)
        self.buttonBox.setCenterButtons(True)
        self.buttonBox.setObjectName("buttonBox")
        self.label = QtWidgets.QLabel(colorDialog)
        self.label.setGeometry(QtCore.QRect(20, 90, 100, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(colorDialog)
        self.label_2.setGeometry(QtCore.QRect(20, 130, 100, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(colorDialog)
        self.label_3.setGeometry(QtCore.QRect(20, 170, 100, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.label_4 = QtWidgets.QLabel(colorDialog)
        self.label_4.setGeometry(QtCore.QRect(20, 210, 100, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.red = QtWidgets.QSpinBox(colorDialog)
        self.red.setGeometry(QtCore.QRect(130, 90, 250, 20))
        font = QtGui.QFont()
        font.setPointSize(10)
        self.red.setFont(font)
        self.red.setMaximum(255)
        self.red.setProperty("value", 220)
        self.red.setObjectName("red")
        self.blue = QtWidgets.QSpinBox(colorDialog)
        self.blue.setGeometry(QtCore.QRect(130, 170, 250, 20))
        self.blue.setMaximum(255)
        self.blue.setProperty("value", 220)
        self.blue.setObjectName("blue")
        self.green = QtWidgets.QSpinBox(colorDialog)
        self.green.setGeometry(QtCore.QRect(130, 130, 250, 20))
        self.green.setMaximum(255)
        self.green.setProperty("value", 220)
        self.green.setObjectName("green")
        self.choose_color = QtWidgets.QComboBox(colorDialog)
        self.choose_color.setGeometry(QtCore.QRect(130, 210, 250, 20))
        font = QtGui.QFont()
        font.setPointSize(9)
        self.choose_color.setFont(font)
        self.choose_color.setObjectName("choose_color")
        self.choose_color.addItem("")
        self.choose_color.addItem("")
        self.choose_color.addItem("")
        self.choose_color.addItem("")
        self.choose_color.addItem("")
        self.choose_color.addItem("")
        self.show_color = QtWidgets.QLabel(colorDialog)
        self.show_color.setGeometry(QtCore.QRect(20, 20, 360, 50))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.show_color.setFont(font)
        self.show_color.setStyleSheet("background-color: rgb(220, 220, 220);")
        self.show_color.setFrameShape(QtWidgets.QFrame.Panel)
        self.show_color.setFrameShadow(QtWidgets.QFrame.Plain)
        self.show_color.setLineWidth(3)
        self.show_color.setAlignment(QtCore.Qt.AlignCenter)
        self.show_color.setObjectName("show_color")

        self.retranslateUi(colorDialog)
        self.buttonBox.accepted.connect(colorDialog.accept)
        self.buttonBox.rejected.connect(colorDialog.reject)
        QtCore.QMetaObject.connectSlotsByName(colorDialog)

    def retranslateUi(self, colorDialog):
        _translate = QtCore.QCoreApplication.translate
        colorDialog.setWindowTitle(_translate("colorDialog", "编辑颜色"))
        self.label.setText(_translate("colorDialog", "红"))
        self.label_2.setText(_translate("colorDialog", "绿"))
        self.label_3.setText(_translate("colorDialog", "蓝"))
        self.label_4.setText(_translate("colorDialog", "预设颜色"))
        self.choose_color.setItemText(0, _translate("colorDialog", "淡灰色"))
        self.choose_color.setItemText(1, _translate("colorDialog", "淡黄色"))
        self.choose_color.setItemText(2, _translate("colorDialog", "浅红色"))
        self.choose_color.setItemText(3, _translate("colorDialog", "淡蓝色"))
        self.choose_color.setItemText(4, _translate("colorDialog", "浅绿色"))
        self.choose_color.setItemText(5, _translate("colorDialog", "亮白色"))
        self.show_color.setText(_translate("colorDialog", "颜色显示"))
